packages/background/src/controllers/NetworkController.ts

```diff
@@ -500,36 +500,36 @@ export default class NetworkController extends BaseController<NetworkControllerS
 
         console.log('get provider', network);
 
-        if (provider) return provider;
-
-        provider = new RPChProvider(
-            'https://primary.gnosis-chain.rpc.hoprtech.net',
-            10000,
-            {
-                discoveryPlatformApiEndpoint: '',
-                entryNodeApiEndpoint: 'http://localhost:13301',
-                entryNodeApiToken: '^^awesomeHOPRr3l4y^^',
-                entryNodePeerId:
-                    '16Uiu2HAm7ZzsLv85xdv5ZjJmUPqxYD2aY86vqDzdH3gCQJ3RBsxb',
-                exitNodePeerId:
-                    '16Uiu2HAmTf3Rfw4Q3pn2T5tTtbhqYZVL4s4cjwVCVsLSJwUTycYL',
-                exitNodePubKey:
-                    '0x036169fb47b14118dda35d866093c4224bcf8f1a1310108f4f2a79eaaddd1c4347',
-                freshNodeThreshold: 1000,
-                maxResponses: 300,
-            },
-            (k, v) => {
-                return new Promise<void>((resolve) => {
-                    rpchStore.set(k, v, resolve);
-                });
-            },
-            (k) => {
-                return new Promise((resolve) => {
-                    rpchStore.get(k, resolve);
-                });
-            }
+        provider = this._getProviderForNetwork(
+            network.chainId,
+            network.rpcUrls[0]
         );
 
+        // If Gnosis, then though RPCh
+        if (networkName == 'xdai') {
+            provider = new RPChProvider(
+                'https://primary.gnosis-chain.rpc.hoprtech.net',
+                {
+                    timeout: 10000,
+                    discoveryPlatformApiEndpoint: 'http://localhost:3020',
+                },
+                (k, v) => {
+                    return new Promise<void>((resolve) => {
+                        rpchStore.set(k, v, resolve);
+                    });
+                },
+                (k) => {
+                    return new Promise((resolve) => {
+                        rpchStore.get(k, resolve);
+                    });
+                }
+            );
+            (provider as RPChProvider).sdk
+                .start()
+                .then(() => console.log('rpch provider started'))
+                .catch((error) => console.log(error));
+        }
+
```

Comment:
I would use `if { .. } else { .. }` so we don't unnecessarily construct two providers